wp-modules/editor/tests/ModelTest.php

```diff
@@ -198,4 +229,55 @@ public function test_slug_and_filename_stay_the_same_after_content_update() {
 		// Make sure the filename does not get changed when only the content is modified.
 		$this->assertSame( 'mismatched-name', $content_modified_pattern['name'] );
 	}
+
+	/**
+	 * Tests that images remain after a pattern is renamed.
+	 */
+	public function test_images_remain_after_a_pattern_is_renamed() {
+		wp_set_current_user( $this->factory()->user->create( [ 'role' => 'administrator' ] ) );
+		add_filter( 'stylesheet_directory', [ $this, 'get_fixtures_directory' ] );
+		do_action( 'init' ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+
+		$content       = get_pattern_by_name( 'a' )['content'];
+		$wp_filesystem = get_wp_filesystem_api();
+
+		// Simulate changing the name of a pattern in the editor.
+		save_metadata_to_pattern_file(
+			null,
+			$this->factory()->post->create(
+				[
+					'post_name' => 'a',
+					'post_type' => get_pattern_post_type(),
+				]
+			),
+			'name',
+			'b'
+		);
+		save_pattern_to_file(
+			$this->factory()->post->create_and_get(
+				[
+					'post_name'    => 'b',
+					'post_title'   => 'B',
+					'post_type'    => get_pattern_post_type(),
+					'post_content' => $content,
+				]
+			)
+		);
+
+		// The image shouldn't have been deleted, as only the pattern name changed.
+		$this->assertTrue(
+			$wp_filesystem->exists( $this->get_fixtures_directory() . '/patterns/images/cup.jpg' )
+		);
+
+		// The previous pattern file should have been deleted.
+		$this->assertFalse(
+			$wp_filesystem->exists( get_patterns_directory() . '/a.php' )
+		);
+
+		// The pattern should have a new file, as its name changed.
+		$this->assertSame(
```

Comment:
This passes even without the [fix](https://github.com/studiopress/pattern-manager/pull/192/files#r1227497186) in this PR.

So I didn't make this test very useful 😄